src/EnergyPlus/HeatBalanceSurfaceManager.cc

```diff
@@ -2651,15 +2683,23 @@ void InitSolarHeatGains(EnergyPlusData &state)
             state.dataSurface->SurfWinSysSolAbsorptance(SurfNum) = 0.0;
         }
     }
-
     if (currSolRadPositive) { // Sun is up, calculate solar quantities
         assert(equal_dimensions(state.dataSurface->SurfReflFacBmToBmSolObs,
                                 state.dataSurface->SurfReflFacBmToDiffSolObs)); // For linear indexing
         assert(equal_dimensions(state.dataSurface->SurfReflFacBmToBmSolObs,
                                 state.dataSurface->SurfReflFacBmToDiffSolGnd)); // For linear indexing
+        Real64 GndReflSolarRad = 0.0;
+        Real64 GndSolarRadInc = max(state.dataEnvrn->BeamSolarRad * state.dataEnvrn->SOLCOS(3) + state.dataEnvrn->DifSolarRad, 0.0);
         for (int SurfNum = 1; SurfNum <= state.dataSurface->TotSurfaces; ++SurfNum) {
             state.dataSurface->SurfSkySolarInc(SurfNum) = state.dataEnvrn->DifSolarRad * state.dataSolarShading->SurfAnisoSkyMult(SurfNum);
-            state.dataSurface->SurfGndSolarInc(SurfNum) = state.dataEnvrn->GndSolarRad * Surface(SurfNum).ViewFactorGround;
+            if (state.dataSurface->UseSurfPropertyGndSurfRefl(SurfNum)) {
+                GndReflSolarRad =
+                    GndSolarRadInc * state.dataSurface->GroundSurfsProperty(state.dataSurface->GroundSurfsPropertyNum(SurfNum)).SurfsReflAvg;
+                state.dataSurface->GndReflSolarRad(SurfNum) = GndReflSolarRad;
+            } else {
+                GndReflSolarRad = state.dataEnvrn->GndSolarRad;
+            }
+            state.dataSurface->SurfGndSolarInc(SurfNum) = GndReflSolarRad * Surface(SurfNum).ViewFactorGround;
```

Comment:
The ground reflected solar radiation for each exterior surface is calculated here and set to `GndReflSolarRad` new surface variable.